docs/tutorials/particle-filtering.md

```diff
@@ -1,3 +1,451 @@
+
+```cpp
+auto motion_model = [&](double particle_position, auto& random_engine) {
+  const double distance = parameters.velocity * parameters.dt;
+  std::normal_distribution<double> distribution(distance, parameters.translation_sd);
+  return particle_position + distribution(random_engine);
+};
+
+
+```
+
+To apply this motion model to all the `particles` range we use `beluga::actions::propagate` function.
+
+```cpp
+particles |= beluga::actions::propagate(std::execution::seq, motion_model);
+```
+
+:::{note}
+The range $x'_t$ represent the posterior $bel'(x'_t)$ before incorporating the measurement $z_t$. This posterior or particle set distribution
+does not match the updated belief $bel(x_t)$ explain in [key concepts](../concepts/key-concepts.md).
+:::
+
+:::{hint}
+The effect of the motion model is to increase the space occupied by the particles.
+:::
+
+### 1.6 Update
+
+We will divide this stage into two steps:
+
+* [Measurement Update](#161-measurement-update)
+* [Resample](#162-resample)
+
+Together, these steps constitute the **Update** phase.
+
+### 1.6.1 Measurement Update
+
+To simulate sensor data, we will create a landmark detector, which is capable of measuring the distance between the robot's `current_position` and nearby landmarks. These measurements are called `detections`, where they consist of an array of distances from the robot's `current_position` to each visible landmark. The range of view of the sensor can be configured by `parameters.sensor_range`.
+
+```cpp
+auto detections =
+      parameters.landmark_map |                                                                            //
+      ranges::views::transform([&](double landmark) { return landmark - current_position; }) |             //
+      ranges::views::remove_if([&](double range) { return std::abs(range) > parameters.sensor_range; }) |  //
+      ranges::to<std::vector>;
+```
+
+* `parameters.landmark_map` is the map generated above.
+* `ranges::views::transform` return the relative position of each landmark to the robot's `current_position`.
+* `ranges::views::remove_if` will remove all landmarks farther out of the sensor's viewing range.
+* `ranges::to<std::vector>` convert the range to a `std::vector<double>`.
+
+<!-- This sensor model, receives as inputs the landmark map, the sensor data and a particle, and outputs the likelihood (or weight) of the sensor data assuming that the particle represent the true state. -->
+
+<!-- TODO(alon): What is it a sensor model explanation should be in another doc like 'extending-beluga' -->
+In this tutorial we will use an approximation to the **landmark sensor model with known correspondence** (_Probabilistic Robotics [^ProbRob], page 149_), since the doors (the landmarks in this case) are indistinguishable from one another, and the are treated as identical in the context of measurement.
+
+For the **landmark sensor model with known correspondence** we will use the following definitions:
```

Comment:
@Alondruck hmm, this model is not the model described in the book.

Reply:
The same thing I posted in the comment on the motion model.

Reply:
Same reply. If there's no book that explains it from first principles, we have to do it ourselves.